### PR TITLE
Click event handler to control resetting the component

### LINKED ISSUE #183: [Bug] BlazoredTypeAhead inside BlazorStrap modal: Dropdown disappears when scrollbar is touched
I have a Blazor Webassembly project running on .Net 5 RC2. I am using a [BlazoredTypeahead ](https://github.com/Blazored/Typeahead) inside a [BlazorStrap ](https://github.com/chanan/BlazorStrap) Modal. When I try to grab the dropdown's scrollbar, the dropdown gets closed. 

I have been unable to determine whether the issue is with BlazoredTypeahead or with BlazorStrap, so I will be cross-posting this item in both the projects.

A reproduction is available here: https://github.com/bancroftway/blazorissuesrepro/tree/typeaheadissues

A video demonstrating the issue is here: https://www.youtube.com/watch?v=m_OftI3jDCQ

**Expected behavior**
Dropdown should not get dismissed when scrollbar is touched.

**Hosting Model (is this issue happening with a certain hosting model?):**
 - Blazor WebAssembly

### PR DESCRIPTION
**Pull request by HAL-NINE-THOUSAND**
Hi Chris,

As requested, a PR for https://github.com/Blazored/Typeahead/issues/183

Resolves #183

### CODE REVIEW

File: src/Blazored.Typeahead/wwwroot/blazored-typeahead.js [hunk 1]
```diff
@@ -16,5 +18,64 @@ window.blazoredTypeahead = {
...
+    onOutsideClick: function (searchTextElement, dotnetRef, methodName, clearOnFire) {
+
+        if (searchTextElement == null) {
+            return;
+        }
+
+        var bId = "";//get the blazor internal ID to distinguish different components
+        for (var clearCount = 0; clearCount < searchTextElement.attributes.length; clearCount++) {
+            var a = searchTextElement.attributes[clearCount];
+            if (a.name[0] == '_') {
```
Comment (chrissainty) [inline src/Blazored.Typeahead/wwwroot/blazored-typeahead.js:53]: Nit pick: Should this be `_bl_` as that is the full prefix Blazor uses for internal IDs?

**Review by chrissainty — CHANGES_REQUESTED** [review]
This looks to be working really well! Nice work @HAL-NINE-THOUSAND 😃. I've asked one little nit pick, see what you think.

When I was testing it I found the Typeahead to but a little sluggish when closing, I thought one of the `Task.Delays` might be causing the issue and I was right. I removed it and everything worked as expected and the speed was much much better. The offending line is 289 inside the `ResetControl` method. With that delay removed the method can be changed to a `void` type as well. 

I would have added this as a code suggestion on the review but I can't make suggestions on code you didn't edit, as far as I can see. Would you be able to make this change as well? Then we can get this merged.

File: src/Blazored.Typeahead/wwwroot/blazored-typeahead.js (hunk 1 above)
Comment (HAL-NINE-THOUSAND) [reply src/Blazored.Typeahead/wwwroot/blazored-typeahead.js]: Yep good point, I just have an aversion to js startsWith for compatibility issues but I think that's an aversion I can now start to let go ^^ - this is included in the new commit

**Comment by HAL-NINE-THOUSAND** [comment]
Hi Chris,

Thanks, I've committed the delay removal the better ID check

**Review by chrissainty — APPROVED** [review]
Looks great! Thank you @HAL-NINE-THOUSAND :shipit:

**Comment by HAL-NINE-THOUSAND** [comment]
Glad I could help Chris, your libraries have been fantastic for learning Blazor and actually getting it out there in the wild

**Pull request merged**